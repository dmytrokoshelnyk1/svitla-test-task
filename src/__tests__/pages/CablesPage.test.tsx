// @ts-nocheck
import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Cables from '@/app/cables/page'
import { jest } from '@jest/globals'
import { fireEvent, waitFor } from '@testing-library/dom'

const mockBackFn = jest.fn()
jest.mock('next/navigation', () => {
  return {
    __esModule: true,
    useRouter: () => ({
      push: jest.fn(),
      replace: jest.fn(),
      prefetch: jest.fn(),
      back: mockBackFn,
    }),
  }
})

describe('Cables page', () => {
  it('Renders correctly initial component', async () => {
    render(<Cables />)
    const title = screen.getByText('Cable types')
    await waitFor(() => expect(screen.getByText('<-')).toBeInTheDocument())
    await waitFor(() => expect(screen.getByText('->')).toBeInTheDocument())
    await waitFor(() =>
      expect(screen.getByTestId('current-page')).toHaveTextContent('1')
    )

    expect(title).toBeInTheDocument()
  })

  it('Should fetch data right after render', async () => {
    const page = render(<Cables />)
    await screen.getByText('Loading...')
  })

  it('Should fetch 2 page after click -> button', async () => {
    render(<Cables />)
    await waitFor(() => {
      expect(screen.getByText('->')).toBeInTheDocument()
    })
    fireEvent.click(screen.getByText('->'))
    await waitFor(() =>
      expect(screen.getByTestId('current-page')).toHaveTextContent('2')
    )
  })

  it('renders correctly toMatchSnapshot', () => {
    const tree = renderer.create(<Cables />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
