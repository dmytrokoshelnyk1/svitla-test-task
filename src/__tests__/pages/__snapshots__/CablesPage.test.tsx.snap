// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cables page renders correctly 1`] = `
<div
  className="pageContainer"
>
  <div
    className="topBlock"
  >
    <button
      className="backButton"
      onClick={[Function]}
      type="button"
    >
      &lt;- Go Back
    </button>
  </div>
  <div
    className="listContainer"
  >
    <div
      className="listHeader"
    >
      <h2>
         Cable types 
      </h2>
    </div>
    <div
      className="listBody"
    >
      <div
        className="container"
      >
        <div
          className="headerContainer"
        >
          <div
            className="columnTitle"
          >
            <span>
              ID
            </span>
          </div>
          <div
            className="columnTitle"
          >
            <span>
              Manufacturer
            </span>
          </div>
          <div
            className="columnTitle"
          >
            <span>
              Diameter
            </span>
          </div>
        </div>
        <div
          className="divider"
        />
      </div>
      <div
        className="paginationContainer"
      >
        <button
          className="buttonNav"
          disabled={true}
          onClick={[Function]}
        >
          &lt;-
        </button>
        <span
          className="currentPage"
          data-testid="current-page"
        >
          1
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cables page renders correctly toMatchSnapshot 1`] = `
<div
  className="pageContainer"
>
  <div
    className="topBlock"
  >
    <button
      className="backButton"
      onClick={[Function]}
      type="button"
    >
      &lt;- Go Back
    </button>
  </div>
  <div
    className="listContainer"
  >
    <div
      className="listHeader"
    >
      <h2>
         Cable types 
      </h2>
    </div>
    <div
      className="listBody"
    >
      <div
        className="container"
      >
        <div
          className="headerContainer"
        >
          <div
            className="columnTitle"
          >
            <span>
              ID
            </span>
          </div>
          <div
            className="columnTitle"
          >
            <span>
              Manufacturer
            </span>
          </div>
          <div
            className="columnTitle"
          >
            <span>
              Diameter
            </span>
          </div>
        </div>
        <div
          className="divider"
        />
      </div>
      <div
        className="paginationContainer"
      >
        <button
          className="buttonNav"
          disabled={true}
          onClick={[Function]}
        >
          &lt;-
        </button>
        <span
          className="currentPage"
          data-testid="current-page"
        >
          1
        </span>
      </div>
    </div>
  </div>
</div>
`;
